src/Media/Audio/AudioPlayer.cpp

```diff
@@ -281,6 +281,14 @@ void AudioPlayer::PlaySound(SoundID eSoundID, int pid, unsigned int uNumRepeats,
         si.sample->Play();
     } else if (pid < 0) {  // exclusive sounds - no override (close chest)
         si.sample->Play();
+    } else if (pid == PID_INVALID) { // initial spell sound, originates from party so must be positioned there
+        provider->SetListenerPosition(pParty->vPosition.x / 50.f,
+                                      pParty->vPosition.y / 50.f,
+                                      pParty->vPosition.z / 50.f);
+        si.sample->SetPosition(pParty->vPosition.x / 50.f,
+                               pParty->vPosition.y / 50.f,
+                               pParty->vPosition.z / 50.f, 500.f);
```

Comment:
Take this outside of the if statements for now so this is default position that can get overridden if valid PID provided.

Reply:
@captainurist has a point. probably better to put this in AudioSample16::Play. If no position provided reset to party position.

Reply:
Should probably drop all the provider->SetListenerPosition calls from this function as well. It gets updated via the game loop anyway.

Reply:
The thing is, in #70 you commented out resetting to zero sound position.
Maybe there was big meaning in this change?

Reply:
I returned it under "!positioned" condition.

Reply:
Sticking plaster fix that I didn't check properly. With that always getting set to zero, all sounds were originating at the party position. So it fixed the 3d sounds (sortof) but broke other bits. Your change is good.